ludwig/schema/features/preprocessing/bag.py

```diff
@@ -54,10 +56,14 @@ class BagPreprocessingConfig(BasePreprocessingConfig):
         parameter_metadata=FEATURE_METADATA[BAG][PREPROCESSING]["lowercase"],
     )
 
-    most_common: int = schema_utils.PositiveInteger(
+    most_common: Union[int, float] = schema_utils.OneOfOptionsField(
```

Comment:
I would add this as a separate parameter instead of overloading this existing one, as to not change the semantics of the pre-existing field. It should be rather straightforward to add a validation check (perhaps as a follow-up PR) to make sure that both parameters aren't explicitly specified at the same time. 

Using a separate parameter also disambiguates what `1` means -- does it mean 100% of the vocabulary or call all other tokens besides the most common as "rare"? And each parameter can be more clearly documented.

Reply:
I agree with making it a separate param, though it will complicate things a little bit, since the default for `most_common` would need to be changed to `None` to allow for specifying `most_common_fraction` (or whatever it's called) without explicitly specifying `most_common=null`.

Reply:
It also means the meaning of `most_common=null` changes from "use all tokens" to "use the default". If the user wanted to use all tokens, they would then need to set `most_common_fraction=1` instead.